core/src/graph/graph.rs

```diff
@@ -245,51 +245,107 @@ impl Graph {
 
 		// Now try to add new connections into pages already being updated
 		// Note: these pages have already been cloned, so we don't clone them again
-		let mut add_iter = ids_to_add.iter();
-		while let Some(_) = add_iter.clone().peekable().peek() {
-			if let Some((_page_id, page)) =
-				updated_pages.iter_mut().find(|(_, page)| !self.is_page_full(&page, false))
-			{
-				let id_to_add = add_iter.next().unwrap(); // safe to unwrap because we peeked above
-				page.add_connection(id_to_add)?;
-			} else {
-				break
+		let mut add_iter = ids_to_add.iter().peekable();
+		for aggressive in vec![false, true] {
+			for page in updated_pages.values_mut() {
+				while let Some(id_to_add) = add_iter.peek() {
+					if let Ok(_) = self.try_add_connection_to_page(
+						page,
+						id_to_add,
+						aggressive,
+						&encryption_key,
+						Some(dsnp_version_config),
+					) {
+						let _ = add_iter.next(); // TODO: prefer advance_by(1) once that stabilizes
```

Comment:
Would you mind explain more about todo?

Reply:
It's just a cleaner way of advancing the iterator if we don't care about the return value. But it's currently unstable nightly.